### PR TITLE
Correlogram, Periodogram: Reimplementation in pyqtgraph

### PR DESCRIPTION
**Pull request by janezd**
##### Issue

Widget uses highcharts, which we don't like.

##### Description of changes

Now it uses pyqtgraph, which we don't like less.

##### Includes
- [X] Code changes
- [X] Tests

### CODE REVIEW

File: orangecontrib/timeseries/widgets/owperiodogram.py [hunk 1]
```diff
@@ -1,123 +1,60 @@
...
+        if attr not in self._cached:
+            if self.data.time_delta.is_equispaced:
+                x = np.ravel(self.data.interp(attr))
+                self._cached[attr] = periodogram_equispaced(x)
+                # TODO: convert periods into time_values-relative values, i.e.
+                # periods *= self.data.time_delta; like lombscargle already does
+                # periods *= self.data.time_delta
+            else:
+                times = np.asanyarray(self.data.time_values, dtype=float)
+                x = np.rave(self.data[:, attr])
```
Comment (ajdapretnar) [inline orangecontrib/timeseries/widgets/owperiodogram.py:30]: I think this should be `x = np.ravel(self.data[:, attr])` 😁

**Comment by ajdapretnar** [comment]
Periodogram issues:
- when using As Timeseries and changing the time attribute to any numeric attribute (say Low from YF), the widget crashes with:
```
----------------------------- TypeError Exception -----------------------------
Traceback (most recent call last):
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owperiodbase.py", line 125, in _selection_changed
    self.replot()
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owperiodogram.py", line 47, in replot
    periods, grams = self.periodogram(attr)
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owperiodogram.py", line 36, in periodogram
    self._cached[attr] = periodogram_nonequispaced(times, x)
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/functions.py", line 167, in periodogram_nonequispaced
    pgram = lombscargle(times, x, freqs)
  File "/Users/ajda/.pyenv-x86/versions/3.9.10/envs/py3.9/lib/python3.9/site-packages/scipy/signal/_spectral_py.py", line 148, in lombscargle
    pgram = _lombscargle(x, y, freqs)
TypeError: Invalid call to pythranized function `_lombscargle(float64[:] (is a view), float64[:], float64[:])'
Candidates are:

    - _lombscargle(float64[:], float64[:], float64[:])

-------------------------------------------------------------------------------
```
Same if one uses instance order.
- Crashes on iris with:
```
-------------------------- AttributeError Exception ---------------------------
Traceback (most recent call last):
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owperiodbase.py", line 125, in _selection_changed
    self.replot()
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owperiodogram.py", line 47, in replot
    periods, grams = self.periodogram(attr)
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owperiodogram.py", line 22, in periodogram
    if self.data.time_delta.is_equispaced:
AttributeError: 'NoneType' object has no attribute 'is_equispaced'
-------------------------------------------------------------------------------
```
- I liked the name of the x axis in the old widget. And thicker lines. I also think the grid in the background helped better determine the height of the bar. I can see lines from multiple attributes overlapped also in the old widget. Hover is probably difficult to implement, so I won't bother you with it. But it was nice to have it in HC and will be useful (if not inevitable) in the future (🤞) Line Chart implementation.

**Comment by ajdapretnar** [comment]
For Correlogram, I'd say similarly to above, I'd vote for thicker lines, grid and x axis name. The widget doesn't crash on the same data as Periodogram, so 👍 .

File: orangecontrib/timeseries/widgets/tests/test_owperiodogram.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# There are not many tests here.
+# Widget adds just a little to PeriodBase, which is tested within
+# tests for OWCorrelogram
+
+
+class TestPeriodogramWidget(WidgetTest):
+    def setUp(self):
+        self.widget: OWPeriodogram = self.create_widget(OWPeriodogram)
+
+    def test_does_something_and_doesnt_crach(self):
```
Comment (ajdapretnar) [inline orangecontrib/timeseries/widgets/tests/test_owperiodogram.py:17]: Kudos on the test name! 👌

**Comment by janezd** [comment]
I think I've done everything except for x-axis title. It is always "Period" - and we know it is period, or else we don't know what the widget is doing. Is it really useful?

I've added horizontal grid. Do we also need the vertical?

Tooltips: yes, implementing them would require mapping coordinates in pyqtgraph, which is black magic. I'll wait for Vesna to finish line chart and then ask for her newly regained expertise. :)

**Comment by ajdapretnar** [comment]
The x axis name would be nice only when exporting the image. I agree it is not strictly necessary in the visualization, because indeed it is always period, but would this be just as evident in a report or a paper?

Horizontal grid suffices. I like it.

Ok, tooltips at a later date. :)

Just one more thing. It would be nice to unify the widget with Line Plot. Specifically in terms of zoom/pan/reset options. This is quite standard in most other Orange widgets.
Screenshot 2022-07-11 at 14 37 27

**Comment by janezd** [comment]
For publishing in a paper, x-axis might have a different name, so let us not add it.

For toolbar, let us this be another issue. I am not very capable at scaling and moving pyqtgraph views. I think @VesnaT is. :) Also: the first button is inapplicable, and zooming would only work horizontally.

**Comment by ajdapretnar** [comment]
You are right, one button is redundant and the other only partially useful. I am just not sure the users will find the small A sign in the lower left corner. 😞 It would be nice to provide a clear way to reset the view. But I agree, having only the reset button would look strange.

**Pull request merged**